src/RecordData/StorageType.h

```diff
@@ -0,0 +1,56 @@
+#ifndef STORAGE_TYPE_H
+#define STORAGE_TYPE_H
+
+namespace mmfs {
+
+class StorageType
+{
+public:
+    virtual ~StorageType() {};
+
+    /**
+     * Initialize the storage device.
+     * @return true if initialization was successful, false otherwise.
+     */
+    virtual bool init() = 0;
+
+    /**
+     * Check if the storage device is ready for read/write operations.
+     * @return true if the storage device is ready, false otherwise.
+     */
+    virtual bool isReady() const = 0;
+
+    /**
+     * Write data to the storage device.
+     * @param data The data to be written.
+     * @param size The size of the data to be written.
+     */
+    virtual void write(char *data, int size) = 0;
```

Comment:
I think to better separate writing strings from random data, we should either `byte *data` or `uint8_t *data` when we have to specify a length, and only use `char *data` when we are asking for a null-terminated string. While essentially identical, it becomes immediately more obvious what is expected of the end user.

Reply:
I just didn't want to bloat `StorageType.h` with a bunch of what can be considered "extra" functions, as then there is more for whoever is extending it to implement. I think the lowest level, such as this class, should have the bare minimum functions needed to use it functionally. 

The `write` method basically is the same as `print`, except I named it to be more in line with typical barebone stream implementations, versus `print` is usually a higher level call. I feel like while `println` may be useful, I don't know if it should be forced for people to implement since they can just call `print(data)` and then `print('\n')`. (unless you think that should be a concrete method in the `StorageType` class, but then it would be slightly inconsistent with the low-level read, write paradgim.

As for the first comment, I can add an overloaded method to not have a size parameter and that be for null-terminated strings, but again I didn't want that higher-level logic to be forced to be implemented in this class, since it is more a quality of life thing than anything else. I also don't know if `byte` exists as a type in all applications, as I think on arduino it's just a proxy for `uint8_t`. If it's not in the PSRAM and SD Card classes though I can add it.